### DISCUSSION TITLE
Mark image as invalid from within the application

### DISCUSSION THREAD

**Discussion by HenryGiraldo** [mcu-tools/mcuboot#1240] [Q&A] [answered]
In a scenario where you want to trigger a rollback from the application, how would you mark an image as invalid?

Scenario: For example, after a DFU, a peripheral stops working because a custom communication protocol has been incorrectly updated. You can't detect this until you start the application.

After some checks, the application could trigger a rollback by marking the running image as invalid. Is there an established way to do that? I can imagine some solutions, but they seem to me to be workarounds, for example modifying the magic number of the image.

**Comment by utzig** [comment]
The old image still exists in the secondary slot after a successful upgrade, so you could just mark it for upgrade again and confirm once it boots.

**Reply by HenryGiraldo** [reply]
How can I mark the old for upgrade from the running application? The running application has newer version and it is running in DIRECT XIP mode.
I think after a reset the newer will run again unless the image is marked as invalid, Am I right?

**Comment by d3zd3z** [comment, marked as answer]
> The old image still exists in the secondary slot after a successful upgrade, so you could just mark it for upgrade again and confirm once it boots.

This would only work if rollback protection is not enabled.

But, to clarify, there isn't really a way to unmark an image as valid after it has been marked as Ok. To do so would require erasing the flash sector, which could result in the loss of the other data in that sector.
